### ISSUE TITLE
C Compiler provided does not support C99 complex under RHEL 8.6 gcc 8.5.0

### ISSUE THREAD

**Issue by PeleDragon** [chetzer-ncpa/ncpaprop-release#27]
I tried to build ncpaprop under RHEL 8.6 with gcc 8.5.0 w/ gcc-toolset-11. I downloaded ncpaprop-release-2.1.0.zip and run 

./configure --with-localpetsc --enable-autodependencies

and got following error

         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
C Compiler provided does not support C99 complex
*******************************************************************************

configure: error: Error running PETSc configure command: ./configure PETSC_ARCH=arch-linux-c-complex PETSC_DIR=PATH_TO/ncpaprop-release-2.1.0/extern/petsc --with-fc=0 --with-debugging=0 --with-mpi=0 --download-f2cblaslapack=1 --with-scalar-type=complex

Similar error when I tried to build via the sources using "git clone https://github.com/chetzer-ncpa/ncpaprop-release.git"   

Is any suggestion to resolve this issue "does not support C99 complex"?

Thanks in advance
Yu-Long

**Comment by chetzer-ncpa** [comment]
Hi Yu-Long

I just saw your other message, so my apologies for the delay on that.

Please send me your configure.log file so I can see if there's any more detailed information in there.

Thanks
-Claus
________________________________
From: PeleDragon ***@***.***>
Sent: Thursday, September 29, 2022 1:20 PM
To: chetzer-ncpa/ncpaprop-release ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [chetzer-ncpa/ncpaprop-release] C Compiler provided does not support C99 complex under RHEL 8.6 gcc 8.5.0 (Issue #27)

[EXTERNAL]


I tried to build ncpaprop under RHEL 8.6 with gcc 8.5.0 w/ gcc-toolset-11. I downloaded ncpaprop-release-2.1.0.zip and run

./configure --with-localpetsc --enable-autodependencies

and got following error

     UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):


________________________________

C Compiler provided does not support C99 complex

________________________________

configure: error: Error running PETSc configure command: ./configure PETSC_ARCH=arch-linux-c-complex PETSC_DIR=PATH_TO/ncpaprop-release-2.1.0/extern/petsc --with-fc=0 --with-debugging=0 --with-mpi=0 --download-f2cblaslapack=1 --with-scalar-type=complex

Similar error when I tried to build via the sources using "git clone https://github.com/chetzer-ncpa/ncpaprop-release.git"

Is any suggestion to resolve this issue "does not support C99 complex"?

Thanks in advance
Yu-Long

**Comment by PeleDragon** [comment]
Dear Claus

Thanks for prompt response.

Here I attached the config.log.

I was able to build ncpaprop via git clone sources under Linux Mint without problem, but no luck under RHEL 8.6.

If you did build it under RHEL 8.6  with gcc version 8.5.0 20210514 (Red Hat 8.5.0-10) (GCC) (or other version of gcc)
please let me know what options I need to set or packages needed  to be download/installed.

Thanks in advance.
Yu-Long

----------------------------------------

**Comment by PeleDragon** [comment]
Dear Claus

I think this is the log file you are asking, under ./extern/petsc

Thanks
Yu-Long

----------------------------------------

**Comment by chetzer-ncpa** [comment]
Dear Yu-Long

I will try to figure this out this week.  The PETSc/SLEPc packages released a new version on 30 September so I am first trying to see if that has caused any problems.

Thanks
-Claus

________________________________

**Comment by chetzer-ncpa** [comment]
Dear Yu-Long,

I have installed RHEL 8.6 with the gcc suite 8.5.0, and I have been unable to reproduce your error.  I have not installed any new yum compiler packages other than the two that are required for ncpaprop, fftw-devel and gsl-devel, neither of which will affect the compilers.  Going through the PETSc setup code, I see that the error is being thrown in 
**${petsc_dir}/config/BuildSystem/config/types.py**, in the function **checkC99Complex**:

```
def checkC99Complex(self):
    '''Check for complex numbers in in C99 std
       Note that since PETSc source code uses _Complex we test specifically for that, not complex'''
    includes = '#include <complex.h>\n'
    body     = 'double _Complex x;\n x = I;\n'
    if not self.checkCompile(includes, body): return    # checkLink can succeed even if checkCompile fails
    if self.checkLink(includes, body):
      # recheck with _GNU_SOURCE active (Issues with crayCC on aarch64)
      includes = '#define _GNU_SOURCE\n#include <complex.h>\n'
      if not self.checkCompile(includes, body): return
      self.addDefine('HAVE_C99_COMPLEX', 1)
      self.c99_complex = 1
    return

```
If I'm reading this correctly, the test that is run is to compile and link the following C program:

```
#include <complex.h>
int main() {
  double _Complex x;
  x = I;
}
```

if there is output or a nonzero return value then the test fails.  On my RHEL system the test succeeds:

```
(base) claus@alien:~/tmp$ cat complextest.c
#include <complex.h>
int main() {
      double _Complex x;
      x = I;
}
(base) claus@alien:~/tmp$ gcc complextest.c
(base) claus@alien:~/tmp$ ./a.out
(base) claus@alien:~/tmp$ echo $?
0
(base) claus@alien:~/tmp$
```

My advice is to try to run the test yourself, manually, by creating the above program, compiling, and running.  If you get any output, that would explain why the configuration test is failing, and the error message should give some insight as to why.

**Comment by PeleDragon** [comment]
Dear Claus,

Many thanks for your going through installation of RHEL 8.6 with gcc 8.5.0 and build it, then finding the complex test.
I found the file you specified and try to compile and run, still with errors.   

Since you are able to ./configure and make through, I decide to login as a different user account with default environment. This time, I am able to ./configure and make it without problem via git clone sources.  It seems the user I tried to build might have incompatible env.

Really appreciate your time and help to resolve this issue.

Thanks
Yu-Long

----------------------------------------
